guvnor-ala/guvnor-ala-build-maven/src/main/java/org/guvnor/ala/build/maven/model/impl/MavenBinaryImpl.java

```diff
@@ -21,41 +21,37 @@
 import org.guvnor.ala.config.CloneableConfig;
 import org.uberfire.java.nio.file.Path;
 
-import static org.uberfire.commons.validation.PortablePreconditions.*;
+import static org.uberfire.commons.validation.PortablePreconditions.checkNotNull;
 
 public class MavenBinaryImpl implements MavenBinary,
                                         CloneableConfig<MavenBinary> {
 
-    private final String type;
-    private final Project sourceProject;
+    private Path path;
 
-    public MavenBinaryImpl( final Project sourceProject ) {
-        this.type = "Maven";
-        this.sourceProject = checkNotNull( "sourceProject", sourceProject );
+    private String name;
+
+    public MavenBinaryImpl(final Path path, final String name) {
+        this.path = checkNotNull( "path", path );
+        this.name = checkNotNull( "name", name );
     }
 
     @Override
     public Project getProject() {
-        return sourceProject;
+        return null;
```

Comment:
@cristianonicolai ok so you decided to create two types.. one for Binaries that have the reference to the project that was used to build it and one for the projects that were resolved used aether right? 
Wondering if we just need to use the same type.. just in case that at some point we want to add the sources references in the future. For now it is ok.